File: Programmers/level1/leastRectangle/leastRectangle.js
// https://programmers.co.kr/learn/courses/30/lessons/86491?language=javascript

export const solution = (sizes) => {
    let w = Math.max(...wh(0, sizes));
    let h = Math.max(...wh(1, sizes))
    return w * h;
}

const wh = (maxmin, arr) => { 
    let temp = []
    if(maxmin === 0){
        arr.forEach(el => {
            temp.push(Math.max(el[0],el[1]))
        })
    } else {
        arr.forEach(el => {
            temp.push(Math.min(el[0],el[1]))
        })
    }
    // console.log(temp)
    return temp;
}

File: Programmers/level1/leastRectangle/least_rectangle.py
# https://programmers.co.kr/learn/courses/30/lessons/86491?language=python3

def solution(sizes):

    w = [max(size[0], size[1]) for size in sizes]
    h = [min(size[0], size[1]) for size in sizes]

    return max(w) * max(h)
